test/EndToEndTests/Tests/Client/Microsoft.OData.Client.E2E.Tests/ActionOverloadingTests/Tests/ActionOverloadingQueryTests.cs

```diff
@@ -0,0 +1,253 @@
+
+using Microsoft.AspNetCore.OData;
+using Microsoft.AspNetCore.OData.Routing.Controllers;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.OData.Client.E2E.TestCommon;
+using Microsoft.OData.Client.E2E.TestCommon.Common;
+using Microsoft.OData.Client.E2E.Tests.ActionOverloadingTests.Server;
+using Microsoft.OData.Client.E2E.Tests.Common.Clients.EndToEnd.Default;
+using Microsoft.OData.Client.E2E.Tests.Common.Server.EndToEnd;
+using Microsoft.OData.Edm;
+using Xunit;
+
+namespace Microsoft.OData.Client.E2E.Tests.ActionOverloadingTests.Tests
+{
+    public class ActionOverloadingQueryTests : EndToEndTestBase<ActionOverloadingQueryTests.TestsStartup>
+    {
+        private readonly Uri _baseUri;
+        private readonly Container _context;
+        private IEdmModel _model = null;
+        public const string PersonTypeName = "Microsoft.OData.Client.E2E.Tests.Common.Server.EndToEnd.Person";
+        public const string EmployeeTypeName = "Microsoft.OData.Client.E2E.Tests.Common.Server.EndToEnd.Employee";
+        public const string SpecialEmployeeTypeName = "Microsoft.OData.Client.E2E.Tests.Common.Server.EndToEnd.SpecialEmployee";
+        public const string ContractorTypeName = "Microsoft.OData.Client.E2E.Tests.Common.Server.EndToEnd.Contractor";
+        public const string ProductTypeName = "Microsoft.OData.Client.E2E.Tests.Common.Server.EndToEnd.Product";
+        public const string OrderLineTypeName = "Microsoft.OData.Client.E2E.Tests.Common.Server.EndToEnd.OrderLine";
+        public const string MetadataPrefix = "$metadata#Default.";
+        public const string ContainerPrefix = "#Default.";
+
+        public class TestsStartup : TestStartupBase
+        {
+            public override void ConfigureServices(IServiceCollection services)
+            {
+                services.ConfigureControllers(typeof(ProductsController), typeof(OrderLinesController), typeof(PeopleController), typeof(MetadataController));
+
+                services.AddControllers().AddOData(opt => opt.Count().Filter().Expand().Select().OrderBy().SetMaxTop(null)
+                    .AddRouteComponents("odata", CommonEndToEndEdmModel.GetEdmModel()));
+            }
+        }
+
+        public ActionOverloadingQueryTests(TestWebApplicationFactory<TestsStartup> fixture)
+            : base(fixture)
+        {
+            _baseUri = new Uri(Client.BaseAddress, "odata/");
+            _context = new Container(_baseUri);
+            _context.HttpClientFactory = HttpClientFactory;
+            _model = CommonEndToEndEdmModel.GetEdmModel();
+        }
+
+        /// <summary>
+        /// Verify metadata of the operations defined in the test service.
+        /// </summary>
+        [Fact]
+        public void QueryMetadataTest()
+        {
+            Dictionary<string, string> expectedUpdatePersonInfoOperations = new Dictionary<string, string>()
+            {
+                { "", null }, // the unbound action

```

Comment:
What do these values mean?

Reply:
These are placeholder values for the expected operations.